File: algorithms/search_algo/C/linear_search.c
#include<stdio.h>


void main()
{
int arr[100],i,n,x,flag=0;
printf("enter the no of elements\n");
scanf("%d",&n);
printf("\nenter the array elments\n");
for(i=0;i<n;i++)
{
scanf("%d",&arr[i]);
}
printf("\nenter the element to be searched\n");
scanf("%d",&x);
for(i=0;i<n;i++)
{
if(arr[i]==x)
	{printf("\nnumber is at pos %d\n",i);
	flag=1;
	}
}
if(flag==0)
	printf("the number is not found");
}

File: algorithms/search_algo/linear_search.py


def linear_search(array,search_num):
    flag = 0
    for i in range(len(array)):
        if array[i] == search_num:
            flag = 1
            break
    if flag == 1:
        print(f"{search_num} is at position {i}")
    else:
        print(f"No such element {search_num}")


if __name__ == "__main__":
    array = input("Enter the elements ")
    array = [int(num) for num in array.split(" ")]
    search_num = int(input("\nEnter the number to be searched "))
    linear_search(array,search_num)